<?xml version="1.0"?>
<launch>
  
  <arg name="joystick" default="true"/>
  <arg name="comms" default="true"/>
  <arg name="use_imu" default="true"/>
  <arg name="imu_device" default="/dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A603UZCU-if00-port0"/>
  <arg name="imu_topic" default="/arduimu_v3/imu"/>
  <arg name="yaw_mixer" default="Bias"/>
  <arg name="siar_device_1" default="/dev/serial/by-id/usb-FTDI_MM232R_USB_MODULE_board2-if00-port0"/>
  <arg name="siar_device_2" default="/dev/serial/by-id/usb-FTDI_MM232R_USB_MODULE_board2-if00-port0"/>
  <arg name="joy_device" default="/dev/serial/by-id/usb-FTDI_FT232R_USB_UART_A6DDSJE-if00-port0"/>
  <arg name="battery_device" default="/dev/serial/by-id/usb-FTDI_MM232R_USB_MODULE_FTGT8JO-if00-port0"/>
  <arg name="two_boards" default="false"/>
  <arg name="perception" default="true"/>
  <arg name="bag" default="true"/>
  <arg name="publish_tf" default="false"/>
  <arg name="obstacle_height" default="0.075"/>
  <arg name="resolution" default ="0.025"/>
  
  <!-- Bring up the raposa node and the imu if desired -->
  <node name="siar_node" pkg="siar_driver" type="siar_node" output="screen">
    <param name="siar_device_1" value="$(arg siar_device_1)"/>
    <param name="siar_device_2" value="$(arg siar_device_2)"/>
    <param name="battery_device" value="$(arg battery_device)"/>
    <param name="yaw_mixer" value="$(arg yaw_mixer)"/>
    <param name="imu_device" value="$(arg imu_device)"/>
    <param name="use_imu" value="$(arg use_imu)"/>
    <param name="freq" value="20"/>
    <param name="joy_device" value="$(arg joy_device)"/>
    <param name="two_boards" value="$(arg two_boards)"/>
    <param name="publish_tf" value="$(arg publish_tf)"/>
  </node>
  <include file="$(find arduimu_v3)/launch/arduimu_v3.launch" if="$(arg use_imu)" >
    <arg name="device" value="$(arg imu_device)"/>
  </include>
  
  <node name="siar_joy" pkg="siar_driver" type="siar_teleop_joy" output="screen" if="$(arg joystick)">
    <remap from="/cmd_vel" to="/user_cmd_vel"/>
    <param name="max_auto_mode" value="2"/>
    <param name="max_joy_time" value="30"/>
  </node>
  <node name="joy_node" pkg="joy" type="joy_node" if="$(arg joystick)" />
  
  <!-- Automatic control -->
   <node name="siar_controller" pkg="siar_controller" type="siar_controller_node">
      <remap from="/cmd_vel_in" to="/user_cmd_vel"/>
      <remap from="/cmd_vel_out" to="/cmd_vel"/>
      <remap from="/altitude_map" to="/costmap_node/costmap"/>
      
      <rosparam file="$(find siar_controller)/cfg/siar_controller.yaml" command="load"  /> 
    </node>
  <include file="$(find siar_costmap)/launch/siar_costmap.launch">
    <arg name="obstacle_height" value="$(arg obstacle_height)"/>
    <arg name="resolution" value="$(arg resolution)"/>
  </include>
  
  <!-- Comms -->
  <include file="$(find siar_driver)/launch/comms.launch" if="$(arg comms)"/>
  
  <!-- Perception -->
  <include file="$(find siar_driver)/launch/perception.launch" if="$(arg perception)"/>
  
  <!-- Bag -->
  <include file="$(find siar_driver)/launch/bag.launch" if="$(arg bag)"/>
</launch>
